script.js

```diff
@@ -1,12 +1,70 @@
+let allEpisodes;
+
 //You can edit ALL of the code here
 function setup() {
-  const allEpisodes = getAllEpisodes();
+  allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
 
+const search = document.getElementById("searchEpisode");
+const rootElem = document.getElementById("root");
+
+search.addEventListener("input",(element) => {
+  const value = element.target.value.toLowerCase();
+  //console.log(value)
+
+
+  const filterEpisodes = allEpisodes.filter((episode) =>
+       episode.name.toLowerCase().includes(value) ||
+        episode.summary.toLowerCase().includes(value)
+        );
+
+        console.log(filterEpisodes);
+
+
+        rootElem.innerHTML ="";
+        makePageForEpisodes(filterEpisodes); //////continue
```

Comment:
I quite like the way you've done this, specifically that you've had makePageForEpisodes accept an array so that you can easily change what it displays using filter and friends to change the orginal array. Formatting is a bit off in places tho